src/resources/embeddings.ts

```diff
@@ -9,9 +9,53 @@ export class Embeddings extends APIResource {
    */
   create(
     body: EmbeddingCreateParams,
-    options?: Core.RequestOptions,
+    options?: Core.RequestOptions<EmbeddingCreateParams>,
   ): Core.APIPromise<CreateEmbeddingResponse> {
-    return this._client.post('/embeddings', { body, ...options });
+    Core.debug('request', 'Sending request with arguments:', { body, ...options });
+
+    const hasUserProvidedEncodingFormat = !!body.encoding_format;
+    let encoding_format: EmbeddingCreateParams['encoding_format'] =
+      hasUserProvidedEncodingFormat ? body.encoding_format : 'base64';
+
+    if (hasUserProvidedEncodingFormat) {
+      Core.debug('Request', 'User defined encoding_format:', body.encoding_format);
+    } else {
+      // No encoding_format specified, defaulting to base64 for performance reasons
+      // See https://github.com/openai/openai-node/pull/1312
+      encoding_format = 'base64';
+    }
+
+    const response = this._client.post('/embeddings', {
+      body: {
+        ...body,
+        encoding_format: encoding_format as EmbeddingCreateParams['encoding_format'],
+      },
+      ...options,
+    });
+
+    // if the user specified an encoding_format, return the response as-is
+    if (hasUserProvidedEncodingFormat) {
+      return response as Core.APIPromise<CreateEmbeddingResponse>;
```

Comment:
nit: do the cast to `Core.APIPromise` on the `response` variable instead of here

Reply:
fixed.